packages/core/src/modules/connections/services/ConnectionService.ts

```diff
@@ -119,9 +119,11 @@ export class ConnectionService {
       imageUrl: imageUrl ?? agentContext.config.connectionImageUrl,
     })
 
+    const parentThreadId = outOfBandRecord.isImplicitInvitation ? invitationDid : outOfBandInvitation.id
```

Comment:
The best thing about that is that it will sweep that ugly `isImplicitInvitation` property under the carpet :stuck_out_tongue: 

But take a look at the new approach, where we have this convenient getter but also removes `isImplicitInvitation` and instead adds the threadId tag to properly identify the OobRecord (only in those cases where implicit invitations are being used). I think it's more compact an clean.